integration/assembly.go

```diff
@@ -69,30 +69,29 @@ func rawMakeEngine(gdb *gossip.Store, cdb *abft.Store, cfg Configs) (*abft.Lache
 	return engine, vecClock, blockProc, nil
 }
 
-func makeEngine(chaindataDir string, cfg Configs) (engine *abft.Lachesis, vecClock *vecmt.Index,
-	gdb *gossip.Store, cdb *abft.Store, blockProc gossip.BlockProc, dbsClose func() error, err error) {
+func makeEngine(chaindataDir string, cfg Configs) (
+	engine *abft.Lachesis,
+	vecClock *vecmt.Index,
+	gdb *gossip.Store,
+	cdb *abft.Store,
+	blockProc gossip.BlockProc,
+	dbsClose func() error,
+	err error,
+) {
 	dbs, err := GetDbProducer(chaindataDir, cfg.DBs.RuntimeCache)
 	if err != nil {
-		return nil, nil, nil, nil, gossip.BlockProc{}, nil, err
+		return
```

Comment:
```suggestion
		return nil, nil, gdb, cdb,  gossip.BlockProc{}, nil, err
```
and similarly for the other returns should work as well, but making explicit that the those 2 variables are still needed.

Reply:
I do not agree with this solution. 
The reason is that you need to reason the returns on each of the return cases, and in each of them the valid solution is to return the actual contents of the variable at the line.  It is extra code, therefore prone to errors.

The real problem with this function that it returns 7 parameters including a lambda.

Reply:
I would also prefer to avoid [naked returns](https://boldlygo.tech/archive/2024-08-02-naked-returns/). It makes it even harder to understand what is returned where.

Reply:
I have to insist here,

the original error only existed because of not using naked returns. Error management during deferred lambdas requires named returns, when using named returns, only naked returns should be used, to avoid un-intentional overwriting of the named variable. which is the original error, which passed review. 

Doing a normal return can very easily overseen, because the assignment to the captured variable is implicit. 
Under the scenario of named returns, only naked returns should be used. precisely to force reviewer to check the correctness of the variables.

Reply:
"Named return values" and "naked returns" are two different things. You can use named return values also with regular returns. The values you are returning then are getting assigned to the named return values before processing the deferred calls.

Reply:
rewrote using _ as name for all not-error returns.